com.unity.ml-agents.extensions/Runtime/Match3/Move.cs

```diff
@@ -0,0 +1,237 @@
+using System;
+
+namespace Unity.MLAgents.Extensions.Match3
+{
+    /// <summary>
+    /// Directions for a Move.
+    /// </summary>
+    public enum Direction
+    {
+        /// <summary>
+        /// Move up (increasing row direction).
+        /// </summary>
+        Up,
+
+        /// <summary>
+        /// Move down (decreasing row direction).
+        /// </summary>
+        Down, // -row direction
+
+        /// <summary>
+        /// Move left (decreasing column direction).
+        /// </summary>
+        Left, // -column direction
+
+        /// <summary>
+        /// Move right (increasing column direction).
+        /// </summary>
+        Right, // +column direction
+    }
+
+    /// <summary>
+    /// Struct that encapsulates a swap of adjacent cells.
+    /// A Move can be constructed from either a starting row, column, and direction,
+    /// or enumerated from 0 to NumPotentialMoves()-1
+    /// </summary>
+    public struct Move
+    {
+        /// <summary>
+        /// Index of the move, from 0 to NumPotentialMoves-1
+        /// Moves are enumerated as the internal edges of the game grid.
+        /// Left/right moves come first. There are (maxCols - 1) * maxRows of these.
+        /// Up/down moves are next. There are (maxRows - 1) * maxCols of these.
+        /// </summary>
+        public int MoveIndex;
+
+        /// <summary>
+        /// Row of the cell that will be moved.
+        /// </summary>
+        public int Row;
+
+        /// <summary>
+        /// Column of the cell that will be moved.
+        /// </summary>
+        public int Column;
+
+        /// <summary>
+        /// Direction that the cell will be moved.
+        /// </summary>
+        public Direction Direction;
+
+        /// <summary>
+        /// Construct a Move from its index and the board size.
```

Comment:
A little more details on this method. Replace `a` with `the` since there is a 1 to 1 mapping between index and move and explain that the index corresponds to the index in the list of all potential moves on a board of size (maxRows, maxCols).

Reply:
I rearranged some of the comments and moved them off of the MoveIndex member and into the class summary. I think between that and the comments on the moveIndex argument, it's fairly clear now.